src/view/configuration/edgeConfigurationsSection.jsx

```diff
@@ -0,0 +1,397 @@
+      orgId,
+      imsAccess
+    }));
+  }
+
+  if (firstPageOfEdgeConfigs.length === 1) {
+    const edgeConfig = firstPageOfEdgeConfigs[0];
+    selectInputMethodState.edgeConfig = edgeConfig;
+
+    const {
+      production,
+      staging,
+      development
+    } = await fetchFirstPageOfEachEnvironmentType({
+      orgId,
+      imsAccess,
+      edgeConfigId: edgeConfig.id
+    });
+
+    if (production.length === 1) {
+      selectInputMethodState.productionEnvironment = production[0];
+    }
+
+    if (staging.length === 1) {
+      selectInputMethodState.stagingEnvironment = staging[0];
+    }
+
+    if (development.length === 1) {
+      selectInputMethodState.developmentEnvironment = development[0];
+    }
+  }
+
+  return selectInputMethodState;
+};
+
+const getFreeformInputMethodStateForExistingInstance = ({
+  instanceSettings
+}) => {
+  return {
+    edgeConfigId: instanceSettings.edgeConfigId ?? "",
+    stagingEdgeConfigId: instanceSettings.stagingEdgeConfigId ?? "",
+    developmentEdgeConfigId: instanceSettings.developmentEdgeConfigId ?? ""
+  };
+};
+
+const getFreeformInputStateForNewInstance = () => {
+  return {
+    edgeConfigId: "",
+    stagingEdgeConfigId: "",
+    developmentEdgeConfigId: ""
+  };
+};
+
+export const bridge = {
+  getInstanceDefaults: async ({ initInfo, isFirstInstance }) => {
+    const {
+      company: { orgId },
+      tokens: { imsAccess }
+    } = initInfo;
+
+    const instanceDefaults = {
+      edgeConfigInputMethod: isFirstInstance
+        ? INPUT_METHOD.SELECT
+        : INPUT_METHOD.FREEFORM,
+      edgeConfigSelectInputMethod: await getSelectInputMethodStateForNewInstance(
+        {
+          orgId,
+          imsAccess
+        }
+      ),
+      edgeConfigFreeformInputMethod: getFreeformInputStateForNewInstance()
+    };
+
+    return instanceDefaults;
+  },
+  getInitialInstanceValues: async ({
+    initInfo,
+    isFirstInstance,
+    instanceSettings
+  }) => {
+    const {
+      company: { orgId },
+      tokens: { imsAccess }
+    } = initInfo;
+
+    const instanceValues = await bridge.getInstanceDefaults({
+      initInfo,
+      isFirstInstance
+    });
+
+    let isSuccessfullyPopulatedForSelectInputMethod = false;
+
+    if (isFirstInstance) {
+      try {
+        instanceValues.edgeConfigSelectInputMethod = await getSelectInputMethodStateForExistingInstance(
+          {
+            orgId,
+            imsAccess,
+            instanceSettings
+          }
+        );
+        instanceValues.edgeConfigFreeformInputMethod = getFreeformInputStateForNewInstance();
+        instanceValues.edgeConfigInputMethod = INPUT_METHOD.SELECT;
+        isSuccessfullyPopulatedForSelectInputMethod = true;
+      } catch (e) {
+        // No nothing. We'll fall back to the freeform input method.
+      }
+    }
+
+    if (!isSuccessfullyPopulatedForSelectInputMethod) {
+      instanceValues.edgeConfigSelectInputMethod = await getSelectInputMethodStateForNewInstance(
```

Comment:
I don't understand this part. If this is the second instance there is no need to fetch configs. Indeed, it will return configs for the wrong IMS org.